operations/mimir-mixin/alerts/ingest-storage.libsonnet

```diff
@@ -19,6 +19,110 @@
             message: '%(product)s {{ $labels.%(per_instance_label)s }} in %(alert_aggregation_variables)s is failing to commit the last consumed offset.' % $._config,
           },
         },
+
+        {
+          alert: $.alertName('IngesterFailedToReadRecordsFromKafka'),
+          'for': '5m',
+
+          // Metric used by this alert is reported by Kafka client on read errors from connection to Kafka.
+          // We use node_id to only alert if problems to the same Kafka node are repeating.
+          // If problems are for different nodes (eg. during rollout), that is not a problem, and we don't need to trigger alert.
+          expr: |||
+            sum by(%(alert_aggregation_labels)s, %(per_instance_label)s, node_id) (rate(cortex_ingest_storage_reader_read_errors_total[1m]))
```

Comment:
the `[1m]` won't work if the metrics are scraped with 1m intervals. Jurarj added soem utils to make this configurable. See this example

https://github.com/grafana/mimir/blob/be893e79cd702047d1d130daf035e00cc705c32d/operations/mimir-mixin/alerts/alertmanager.libsonnet#L83-L85

can you change the intervals of the alerts to use `$.alertRangeInterval(1)` for `1m`  `$.alertRangeInterval(5)` for `5m`, etc?

Reply:
I really don't want to use longer interval here, as I expect this is typically a problem that disappears quickly. Also I've opted for quite short `for: 5m` here, and if we used longer range window, that won't work.

Let's first use this, and learn if it works, I expect we will adjust these alerts few times before considering them "ready for everyone".

Reply:
sure, as long as we don't forget 👍

Reply:
So I updated our mixin today finally wanting to enable the validation in our repo that no queries have too short range interval and ran into this :( . I currently point the mixin at top of main, if these alerts are not yet recommended for general consumption, what would be a recommendation with the mixins, to point at tags instead of top of main? Or could we add an option to exclude the alerts which are not yet ready?

Reply:
I'm happy to add an option to exclude these alerts. They are for a new Kafka-based ingestion mode we're working on, and it's far from ready. We added alerts because we plan to start testing it internally.

Would you send a PR with option?

Reply:
Will do thank you.

Reply:
Finally got to it, sorry for the delay. I raised https://github.com/grafana/mimir/pull/7867 to add the option to exclude the alerts.